navigationrail/navigationrail/src/main/java/com/microsoft/device/dualscreen/navigationrail/NavigationRailView.kt

```diff
@@ -0,0 +1,522 @@
+import com.microsoft.device.dualscreen.utils.wm.isFoldingFeatureHorizontal
+import com.microsoft.device.dualscreen.utils.wm.isInDualMode
+import com.microsoft.device.dualscreen.utils.wm.locationOnScreen
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+const val MARGIN_LOWERING_FACTOR = 0.95
+
+/**
+ * A sub class of the [NavigationRailView] that can position its children in different ways when the application is spanned on both screens.
+ * Using the [arrangeButtons] and [setMenuGravity] the children can be split in any way between the two screens.
+ * Animations can be used when changing the arrangement of the buttons on the two screen.
+ */
+class NavigationRailView : NavigationRailView {
+    constructor(context: Context) : super(context) {
+        this.registerWindowInfoFlow()
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        this.registerWindowInfoFlow()
+        extractAttributes(context, attrs)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        this.registerWindowInfoFlow()
+        extractAttributes(context, attrs)
+        menuGravity = Gravity.TOP
+    }
+
+    private var screenMode = ScreenMode.DUAL_SCREEN
+    private var job: Job? = null
+    private var windowLayoutInfo: WindowLayoutInfo? = null
+
+    private var topBtnCount: Int = -1
+    private var bottomBtnCount: Int = -1
+
+    private fun normalizeFoldingFeatureRectForView(): Rect {
+        return windowLayoutInfo.extractFoldingFeatureRect().apply {
+            offset(-locationOnScreen.x, 0)
+        }
+    }
+
+    private fun registerWindowInfoFlow() {
+        val activity = (context as? ComponentActivity)
+            ?: throw RuntimeException("Context must implement androidx.activity.ComponentActivity!")
+        job = activity.lifecycleScope.launch(Dispatchers.Main) {
+            activity.lifecycle.repeatOnLifecycle(Lifecycle.State.RESUMED) {
+                WindowInfoTracker.getOrCreate(activity)
+                    .windowLayoutInfo(activity)
+                    .collect { info ->
+                        windowLayoutInfo = info
+                        onInfoLayoutChanged()
+                    }
+            }
+        }
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        job?.cancel()
+    }
+
+    private fun onInfoLayoutChanged() {
+        setScreenParameters()
+
+        val changeBounds: Transition = ChangeBounds()
+        changeBounds.duration = 300L
+        changeBounds.interpolator = PathInterpolatorCompat.create(0.2f, 0f, 0f, 1f)
+        TransitionManager.beginDelayedTransition(this@NavigationRailView, changeBounds)
+        requestLayout()
+    }
+
+    private var onSwipeListener: OnVerticalSwipeListener =
```

Comment:
The usage of this functionality is indeed pretty cool, but it's completely hidden in the UI. I think we should add something in the sample app to make it easier to be discovered. Even if it's a simple text explaining the feature, could be useful.

Reply:
I agree but this is also true for BottomNavigationView and TabLayout. 
We should discuss a way that will work for all 3 views.